tests/api/lib/indexing_pipeline_test.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+from haystack import Pipeline, Document
+from haystack.components.preprocessors import DocumentSplitter
+from haystack.components.writers import DocumentWriter
+from haystack_integrations.components.embedders.fastembed import FastembedDocumentEmbedder
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+from mockito import mock, when, verify, any
+from mockito.matchers import eq, captor
+
+from search_backend.api.lib.indexing_pipeline import IndexingPipeline
+
+
+class TestIndexingPipeline(unittest.TestCase):
+    def setUp(self):
+        self.mock_document_store = mock(OpenSearchDocumentStore)
+
+        self.mock_pipeline = mock(Pipeline)
+        when(self.mock_pipeline).add_component(any(), any())
+        when(self.mock_pipeline).connect(any(), any())
```

Comment:
Pytest collection is extremely slow. I'd suggest marking these tests as expensive so that they do not run by default unless specifically asked for.  To do this:
In pyproject.toml:

```.toml
# `pytest` configurations
[tool.pytest.ini_options] markers = ["expensive: tests that are costly.", ]
```

In the test:

`@pytest.mark.expensive`
(apply it to a class or test as needed)

To ensure the expensive tests are run when invoking `pytest`:

`pytest -m "expensive"`

Unmarked tests will not be run.

To deselect marked tests:

`pytest -m "not expensive"`

You can include operators like `or` and `and` to control execution of several marks.

Reply:
I didn't notice them being especially slow but perhaps I wasn't paying attention enough.

Reply:
Also, if we don't run these by default, there won't be any unit tests still :) Most of the classes in this library will reference haystack and be subject to the same problem, unless we hide the haystack implementation (e.g. behind our own adapter classes).

Reply:
yes that may be disproportionate for now. It's taking in the region of 18s to execute on my machine, which is curious. I would suggest living with it if it's a 'me' problem though.